connection.go

```diff
@@ -133,6 +134,57 @@ func (c *Connection) Connect() error {
 	return nil
 }
 
+// Accept accepts a connection to the server using the configured address and should be used when the other party
+// is responsible for establishing the connection.
+func (c *Connection) Accept() error {
+	listener, err := net.Listen("tcp", c.addr)
+	if err != nil {
+		return fmt.Errorf("creating listener %s: %w", c.addr, err)
+	}
+
+	c.wg.Add(1)
+	go func() {
+		defer func() {
+			c.wg.Done()
+		}()
+
+		chConn := make(chan net.Conn, 1)
+		chErr := make(chan error, 1)
+
+		go func() {
+			conn, err := listener.Accept()
```

Comment:
Does this handle multiple connections / messages? I can see it handling an initial connection, but once the first message is processed I see `Accept(..)` returning.

Reply:
I can see calling code wrapping this with `go func() { for .Accept() }` in an endless loop, which makes sense to me.

Reply:
this accepts a single connection, once established, the connection remains open for traffic until it is closed

Reply:
There is a good example how `http.Serve` works: https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/net/http/server.go;l=3039;drc=0e90ba933565ab6e719d6981d8d6195e06625f47

in short for each `rw: = ln.Accept()` it creates new connection: `c := srv.newConn(rw)`

in our case, we need a server that will accept new connections and each such connection should be wrapped into `Connection`. I would not put the server functionality into connection.

We already have it implemented: https://github.com/moov-io/iso8583-connection/blob/implement-listen/server/server.go#L45

Maybe we can use it and improve it by adding what's missed?

Reply:
Right so the additional for loop can be outside of this repository to give servers more control over that handling.